cpp/src/arrow/compare.cc

```diff
@@ -1563,6 +1563,9 @@ bool ArrayStatisticsEqualsImpl(const ArrayStatistics& left, const ArrayStatistic
          left.distinct_count == right.distinct_count &&
          left.is_min_exact == right.is_min_exact &&
          left.is_max_exact == right.is_max_exact &&
+         left.is_average_byte_width_exact == right.is_average_byte_width_exact &&
```

Comment:
Could you use the same order in `.h`?

```suggestion
         left.is_average_byte_width_exact == right.is_average_byte_width_exact &&
         left.is_min_exact == right.is_min_exact &&
         left.is_max_exact == right.is_max_exact &&
```